pkg/controller/handlers/mcpcatalog/mcpcatalog.go

```diff
@@ -239,42 +242,155 @@ func (h *Handler) readMCPCatalog(catalogName, sourceURL string) ([]client.Object
 		}
 
 		if err := validation.ValidateCatalogEntryManifest(entry); err != nil {
-			return nil, fmt.Errorf("failed to validate catalog entry %s: %w", entry.Name, err)
+			errs = append(errs, fmt.Errorf("failed to validate catalog entry %s: %w", entry.Name, err))
+			continue
 		}
 		catalogEntry.Spec.Manifest = entry
 
 		objs = append(objs, &catalogEntry)
 	}
 
-	return objs, nil
+	return objs, errors.Join(errs...)
 }
 
-func (h *Handler) readMCPCatalogDirectory(catalog string) ([]types.MCPServerCatalogEntryManifest, error) {
-	files, err := os.ReadDir(catalog)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read catalog directory %s: %w", catalog, err)
+func readMCPCatalogDirectory(catalog string) ([]types.MCPServerCatalogEntryManifest, error) {
```

Comment:
This was just moved from the other file, essentially untouched (I renamed a variable).